.circleci/workflows.yml

```diff
@@ -1343,7 +1352,7 @@ jobs:
     resource_class: << parameters.resource_class >>
     steps:
       - update_known_hosts
-      - checkout
+      - shallow_checkout
```

Comment:
@nalandial I recently made [this change](https://github.com/cypress-io/cypress/pull/28230/files) and we had to back it out for two reasons 😢 
1. contributor PRs - [example failure](https://app.circleci.com/pipelines/github/cypress-io/cypress/57994/workflows/67bea44c-1f52-42e0-b131-6470452bbb92/jobs/2408480) -- forked PRs have the CIRCLE_BRANCH set to `pull/NUM` which isn't a valid branch to checkout. I wrote [this up,](https://github.com/cypress-io/cypress/compare/contributor-prs) but wasn't able to verify;
2. the release-readiness jobs expects the git history to validate the changelog - [example failure](https://app.circleci.com/pipelines/github/cypress-io/cypress/58023/workflows/5cfd219d-e0ec-442c-adac-99e58f362f99/jobs/2409853). This would need a little tweaking to pull down the history on that job specifically.

Also, did you verify this works on Windows? I wasn't able to get the shallow checkout to work on Windows.